### PR TITLE
out_datadog: implement attributes to tags remapping

### PR DESCRIPTION
**Pull request by joyzhang-dd**
implement attributes to tags remapping

### CODE REVIEW

File: plugins/out_datadog/datadog.c [hunk 1]
```diff
@@ -157,10 +189,18 @@ static int datadog_format(const void *data, size_t bytes,
         }
 
         /* Append initial object k/v */
-        int i = 0;
+        int i = 0, ind;
```
Comment (clamoriniere) [inline plugins/out_datadog/datadog.c]: looking at the code style convention you should set `ind = 0;`

**Comment by clamoriniere** [comment]
After some the tests, this metadata remapping works. 
Only some few change to do on the split/remapping

* `cluster_name` should contains only the last part of the input. 
input: `ecs_cluster: arn:aws:ecs:us-east-1:601427279990:cluster/test-fluentbit`
ouput: `cluster_name:test-fluentbit`

* `container_name` the plugin should remove the initial "/"
input: `container_name: /ecs-ced-firelens-test-8-redis-9cbae2cf9bbfe3916300`
output: `container_name:ecs-ced-firelens-test-8-redis-9cbae2cf9bbfe3916300`

* `task_arn` should only take the last part of the input
input: `ecs_task_arn:arn:aws:ecs:us-east-1:601427279990:task/afab6505-2937-42d2-ba8f-889019a3c7b0`
ouput: `task_arn:afab6505-2937-42d2-ba8f-889019a3c7b0`

**Comment by clamoriniere** [comment]
other change:
* rename `task_name` to `task_family`

File: plugins/out_datadog/datadog_remap.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+#include "datadog_remap.h"
+
+const char* ARN_PREFIX = "arn:aws:ecs:";
+
+static void dd_remap_append_kv_to_ddtags(const char* key, const char* val, size_t val_len, flb_sds_t dd_tags)
+{
+        flb_sds_cat(dd_tags, key, strlen(key));
+        flb_sds_cat(dd_tags, ":", 1);
+        flb_sds_cat(dd_tags, val, val_len);
+        flb_sds_cat(dd_tags, ",", 1);
```
Comment (ajacquemot) [inline plugins/out_datadog/datadog_remap.c]: nits, do we want to append `,` first if `dd_tags length > 0` ?

**Comment by gaetan-deputier** [comment]
Overall LGTM ! Nice!

File: plugins/out_datadog/datadog.c (hunk 1 above)
Comment (joyzhang-dd) [reply plugins/out_datadog/datadog.c]: changed

File: plugins/out_datadog/datadog_remap.c (hunk 2 above)
Comment (joyzhang-dd) [reply plugins/out_datadog/datadog_remap.c]: Yes. I was thinking about saving a if-branching on every append, but your way makes code look cleaner. So I changed

File: plugins/out_datadog/datadog_conf.c [hunk 3]
```diff
@@ -108,6 +108,14 @@ struct flb_out_datadog *flb_datadog_conf_create(struct flb_output_instance *ins,
         ctx->dd_message_key = flb_sds_create(tmp);
     }
 
+    tmp = flb_output_get_property("provider", ins);
+    if (tmp && (strncmp(tmp, FLB_DATADOG_REMAP_PROVIDER, strlen(tmp)) == 0)) {
```
[the remark this answers is not in the document]
Comment (joyzhang-dd) [reply plugins/out_datadog/datadog_conf.c]: Yes. Changed.

**Pull request closed without merging**